onnxruntime/core/providers/cpu/ml/tree_ensemble_classifier.cc

```diff
@@ -358,20 +358,23 @@ common::Status TreeEnsembleClassifier<T>::Compute(OpKernelContext* context) cons
   int64_t N = x_dims.size() == 1 ? 1 : x_dims[0];
   Tensor* Y = context->Output(0, TensorShape({N}));
   auto* Z = context->Output(1, TensorShape({N, class_count_}));
-
-  int64_t zindex = 0;
   const T* x_data = X.template Data<T>();
 
-  // for each class
-  std::vector<float> scores;
-  scores.reserve(class_count_);
+  common::Status status;
+#ifdef USE_OPENMP
+#pragma omp parallel for
```

Comment:
Can you change this code to use the thread pool's ParallelFor instead? We're supposed to be moving away from OpenMP in favor of the thread pool.

Reply:
Well, here is what I have with thread pool(size 32):
{"cat" : "Node","pid" :23508,"tid" :23508,"dur" :**94724**,"ts" :1086067,"ph" : "X","name" :"TreeEnsembleClassifier_kernel_time","args" : {"provider" : "CPUExecutionProvider","op_name" : "TreeEnsembleClassifier"}},
As you can see it's much worse.
Of course we could SPEND TIME to investigate what is the best option for thread pool size for a particular a machine. But shall we asking clients to use the same machine?
Further, the code is not neater with the thread pool - it is counter intuitive.